### PR TITLE
Feature/slab component corners shadows

### PR DESCRIPTION
**Pull request by emiliema**
## Description

[SWED-1986](https://payexjira.atlassian.net/browse/SWED-1986)
Slab default and arrow components and the documentation has been updated.

## Motivation and Context

The designers found the old version a bit outdated. By updating the component it will be adjusted to the new UI look and feel of the newest components. The documentation has far less content, therefore much more freedom to personify the component.

## How Has This Been Tested?

Test file has been updated, user tested a bit the responsivity. 
**_Not sure how this update will affect the existing use of this component_**

## Screenshots (if appropriate):
image
image
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have updated the **CHANGELOG** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Review instructions
[Review instructions](../REVIEW_INSTRUCTIONS.md)

### CODE REVIEW

File: src/App/ComponentsDocumentation/components/Slab/index.js [hunk 1]
```diff
@@ -1,210 +1,66 @@
 import React, { Component } from "react";
 
 import { ComponentPreview, DocContainer } from "@docutils";
+
 import CodeTags from "@components/CodeTags";
 
+import { overviewSlabs } from "./constants";
 import { validation } from "@src/scripts/main";
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Slab/index.js:8]: This validation thing is old I think.. Remove it all, also the useEffect longer down in the file.

File: src/App/ComponentsDocumentation/components/Slab/constants.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React from "react";
+
+const Slab = ({ darkShadow, noShadow }) => (
+    <div id="slab-doc"className={`slab slab-default${darkShadow ? " dark" : ""}${noShadow ? " flat" : ""}`}/>
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Slab/constants.js:4]: Space between id="" and className

File: src/App/ComponentsDocumentation/components/Slab/constants.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                                    darkShadow: false,
+                                    noShadow: true
+                                }
+                            }
+                        ]
+                    }
+                ]
+            },
+            title: "Overview slabs",
+            description: "Slab component is basically a content frame which is used as a graphic element background in order to display different types of content on it surface. Slab component uses drop shadows to be more visible and noticeable on the UI. Drop shadows can be an amazing way to add depth and emphasis to your designs, creating an image of floating UI elements. Shadows are a way of creating a container for content without making it feel boxed in."
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Slab/constants.js:44]: Some typos here.. Use this instead 
"The slab component is basically a content frame that is used as a graphic element background in order to display different types of content on its surface. The slab component uses drop shadows to be more visible and noticeable on the UI. Drop shadows can be an amazing way to add depth and emphasis to your designs, creating an image of floating UI elements. Shadows are a way of creating a container for content without making it feel boxed in."

File: src/App/ComponentsDocumentation/components/Slab/index.js [hunk 4]
```diff
@@ -1,210 +1,66 @@
...
-        <h2 id="slab-well">Slab well</h2>
-        <ComponentPreview language="html" showCasePanel codeFigure>
-            <div className="slab slab-default slab-well">
-                Look ma, im in a well!
-            </div>
-        </ComponentPreview>
-    </>
+        </div>
+        <br/>
+        <div> When using a smaller slab in a slab component, in that case the inner slab needs to adjust its border radius accordingly in order the padding between the slabs to remain relevant and consistent.</div>
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Slab/index.js:41]: Weird sentence.. Use this
"When using a smaller slab in a slab component, in that case, the inner slab needs to adjust its border radius accordingly in order for the padding between the slabs to remain relevant and consistent"

File: src/App/ComponentsDocumentation/components/Slab/index.js [hunk 5]
```diff
@@ -1,210 +1,66 @@
...
-                        </ul>
+const basename = process.env.basename;
+
+const OverviewSlab = () => (
+    <section id="slab-doc">
+        <h2 id="default-slab">Overview</h2>
+        <ComponentPreview language="html" showCasePanel showCasePanelAdvanced={overviewSlabs} codeFigure/>
+        <h3>Things to consider</h3>
+        <ul className="list list-bullet">
+            <li>When using the slab component with a combination of other elements on it, we need to consider how the border radius of the slab it is affecting the border radius of the element which takes place as content in the slab.</li>
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Slab/index.js:18]: border-radius

File: src/App/ComponentsDocumentation/components/Slab/index.js [hunk 6]
```diff
@@ -1,210 +1,66 @@
...
+            <li>When using a slab component, the border radius could differ for different devices and sizes. Border radius always requires to be adjusted accordingly to the dimensions.</li>
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Slab/index.js:19]: Border-radius

File: src/App/ComponentsDocumentation/components/Slab/index.js [hunk 7]
```diff
@@ -1,210 +1,66 @@
...
-                Look ma, im in a well!
-            </div>
-        </ComponentPreview>
-    </>
+        </div>
+        <br/>
+        <div> When using a smaller slab in a slab component, in that case the inner slab needs to adjust its border radius accordingly in order the padding between the slabs to remain relevant and consistent.</div>
+        <h3>New design</h3>
+        <p className="list">
+            This is a breaking change regarding the visuals of the component.
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Slab/index.js:44]: 🦸‍♀️ 💯 👌

File: src/App/ComponentsDocumentation/components/Slab/index.test.js [hunk 8]
```diff
@@ -17,37 +17,13 @@ describe("Components: Slab", () => {
         expect(console.warn).toHaveBeenCalled();
     });
 
-    describe("DefaultSlab", () => {
+    describe("OverviewSlab", () => {
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Slab/index.test.js:20]: Create a similar test for SlabArrow

Comment (emiliema) [reply src/App/ComponentsDocumentation/components/Slab/index.test.js:20]: There is already a SlabArrow test

File: src/App/ComponentsDocumentation/components/Slab/constants.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+                                    darkShadow: false,
+                                    noShadow: true
+                                }
+                            }
+                        ]
+                    }
+                ]
+            },
+            title: "Overview slabs",
+            description: "The slab component is a content frame that is used as a graphic element background in order to display different types of content on its surface. By adding drop shadows it gets more noticeable and adds depth and emphasis to the design, creating an image of floating UI elements."
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Slab/constants.js:44]: Straight to the point, veryy good 💯

File: src/App/ComponentsDocumentation/components/Slab/index.js [hunk 10]
```diff
@@ -1,237 +1,76 @@
...
-                        </ul>
+const basename = process.env.basename;
+
+const OverviewSlab = () => (
+    <section id="slab-doc">
+        <h2 id="default-slab">Overview</h2>
+        <ComponentPreview language="html" showCasePanel showCasePanelAdvanced={overviewSlabs} codeFigure/>
+        <h3>Things to consider</h3>
+        <ul className="list list-bullet">
+            <li>When using the slab component with a combination of other elements on it, we need to consider how the border-radius of the slab it is affecting the border-radius of the element which takes place as content in the slab.</li>
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Slab/index.js:15]: this sentence tho.... 
I completely rewrote it.. haha, is this ok? 
"When combining other components with the slab, one must consider how the slab's border-radius affects the edges of the on-top placed component. "

File: src/App/ComponentsDocumentation/components/Slab/index.test.js (hunk 8 above)
Comment (eskilsen) [reply src/App/ComponentsDocumentation/components/Slab/index.test.js:19]: `describe("SlabArrow", () => {
expect(SlabArrow).toBeDefined();
});

1. it("renders", () => {
2. const wrapper = shallow(<SlabArrow />});`

**Review by eskilsen — APPROVED** [review]
image

**Pull request merged**